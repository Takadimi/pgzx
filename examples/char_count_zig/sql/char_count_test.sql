CREATE EXTENSION char_count_zig;

SELECT char_count_zig('aaabccd', 'a');
SELECT char_count_zig('aaabccd', 'b');
SELECT char_count_zig('aaabccd', 'c');
SELECT char_count_zig('aaabccd', 'd');
SELECT char_count_zig('aaabccd', 'e');
SELECT char_count_zig('aaabccd', 'abc');
SELECT char_count_zig('aaabccd', NULL);
SELECT char_count_zig(NULL, 'a');
SELECT char_count_zig(NULL, NULL);
