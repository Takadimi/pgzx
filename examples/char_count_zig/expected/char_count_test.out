CREATE EXTENSION char_count_zig;
SELECT char_count_zig('aaabccd', 'a');
INFO:  input_text: aaabccd

INFO:  target_char: a

INFO:  Target char len: 1

 char_count_zig 
----------------
              3
(1 row)

SELECT char_count_zig('aaabccd', 'b');
INFO:  input_text: aaabccd

INFO:  target_char: b

INFO:  Target char len: 1

 char_count_zig 
----------------
              1
(1 row)

SELECT char_count_zig('aaabccd', 'c');
INFO:  input_text: aaabccd

INFO:  target_char: c

INFO:  Target char len: 1

 char_count_zig 
----------------
              2
(1 row)

SELECT char_count_zig('aaabccd', 'd');
INFO:  input_text: aaabccd

INFO:  target_char: d

INFO:  Target char len: 1

 char_count_zig 
----------------
              1
(1 row)

SELECT char_count_zig('aaabccd', 'e');
INFO:  input_text: aaabccd

INFO:  target_char: e

INFO:  Target char len: 1

 char_count_zig 
----------------
              0
(1 row)

SELECT char_count_zig('aaabccd', 'abc');
ERROR:  Target char is more than one byte
SELECT char_count_zig('aaabccd', NULL);
ERROR:  Unexpected error: UnexpectedNullValue
SELECT char_count_zig(NULL, 'a');
ERROR:  Unexpected error: UnexpectedNullValue
SELECT char_count_zig(NULL, NULL);
ERROR:  Unexpected error: UnexpectedNullValue
